src/components/BetCreatorForm.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react'
+import BetEntry from '../models/BetEntry'
+import { BetDataContext } from '../providers/BetDataProvider'
+
+const newId = -1
+interface Props {
+  onBetAdded: (bet: BetEntry) => void
+}
+const BetCreatorForm: React.FC<Props> = ({ onBetAdded }) => {
+  const [request, setRequest] = React.useState<BetEntry>(
+    new BetEntry(newId, '', '', '', '')
+  )
+  const betConsumer = React.useContext(BetDataContext)
+
+  const addBet = async () => {
+    try {
+      const response = await betConsumer.addNewBet(request)
+      onBetAdded(response)
+    } catch (e: any) {
+      console.error('failed')
+    }
+  }
+  return (
+    <div className="form flex space-between">
+      <textarea
+        placeholder="Enter a bet title"
+        id="bet-title"
+        onChange={(e) => setRequest({ ...request, title: e.target.value })}></textarea>
```

Comment:
I think that we should change our aproach to input updates
This guy explain how to change from rendering whole component every time we type any letter into textarea. [Tutorial](https://youtu.be/GGo3MVBFr1A?t=72).

I think that is good ocassion to check it in real life example.

Reply:
Maybe next time